File: python3/1658.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        s = sum(nums)
        s -= x
        if s < 0:
            return -1
        l, r = 0, 0
        ans, cnt = -1, 0
        n = len(nums)
        while r < n:
            cnt += nums[r]
            while cnt > s:
                cnt -= nums[l]
                l += 1
            if cnt == s:
                ans = max(ans, r - l + 1)
            r += 1
        return n - ans if ans != -1 else -1

File: cpp/1658.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int s = accumulate(nums.begin(), nums.end(), 0);
        s -= x;
        if (s < 0) {
            return -1;
        }
        int l = 0, r = 0;
        int ans = -1, cnt = 0;
        int len = nums.size();
        while (r < len) {
            cnt += nums[r];
            while (cnt > s) {
                cnt -= nums[l];
                ++l;
            }
            if (cnt == s) {
                ans = max(ans, r - l + 1);
            }
            ++r;
        }
        return ans == -1 ? -1 : len - ans;
    }
};